### PR TITLE
Enhance WeightOnlyLinear capability

### PR DESCRIPTION
**Pull request by xin3he**
## Type of Change

Enhance WeightOnlyLinear capability

## Description

support compress_dim, compress_bits and to_half

## Expected Behavior & Potential Risk

UT pass

## How has this PR been tested?

local test

## Dependency Change?

N/A

### CODE REVIEW

File: neural_compressor/model/torch_model.py [hunk 1]
```diff
@@ -394,9 +394,26 @@ def export(
...
-    def convert(self, weight_only=True, weight_config_path=None):
-        """Convert Lineat to WeightOnlyLinear for low memory inference."""
+    def convert(self, weight_only=True, **kwargs):
+        """Convert Lineat to WeightOnlyLinear for low memory inference.
+
+        Args:
+            weight_only (bool, optional): Whether convert model to WeightOnlyLinear. 
+                                          Defaults to True.
+            if weight_only, kwargs includes:
+                weight_config_path (str, optional): Path of weight_config.json. Defaults to None.
```
Comment (hshen14) [inline neural_compressor/model/torch_model.py:404]: weight_quant_config_path?

File: neural_compressor/model/torch_model.py [hunk 2]
```diff
@@ -394,9 +394,26 @@ def export(
...
+                full_range (bool, optional): Whether leverage the last bit of symmetric dtype.
```
Comment (hshen14) [inline neural_compressor/model/torch_model.py:405]: sym_full_range?

File: neural_compressor/model/torch_model.py [hunk 3]
```diff
@@ -394,9 +394,26 @@ def export(
...
+        """Convert Lineat to WeightOnlyLinear for low memory inference.
+
+        Args:
+            weight_only (bool, optional): Whether convert model to WeightOnlyLinear. 
+                                          Defaults to True.
+            if weight_only, kwargs includes:
+                weight_config_path (str, optional): Path of weight_config.json. Defaults to None.
+                full_range (bool, optional): Whether leverage the last bit of symmetric dtype. 
+                                            Defaults to False.
+                compression_factor (str, optional): Path of weight_config.json. Defaults to 32.
```
Comment (hshen14) [inline neural_compressor/model/torch_model.py:407]: better term?

File: neural_compressor/model/torch_model.py [hunk 4]
```diff
@@ -394,9 +394,26 @@ def export(
...
+                compression_dim (str, optional): Path of weight_config.json. Defaults to 'K'.
+                to_half (bool, optional): Set float32 tensor to float16. Defaults to False.
```
Comment (hshen14) [inline neural_compressor/model/torch_model.py:409]: scale_dtype?

File: neural_compressor/model/torch_model.py [hunk 5]
```diff
@@ -394,9 +394,26 @@ def export(
...
+
+        Args:
+            weight_only (bool, optional): Whether convert model to WeightOnlyLinear. 
+                                          Defaults to True.
+            if weight_only, kwargs includes:
+                weight_config_path (str, optional): Path of weight_config.json. Defaults to None.
+                full_range (bool, optional): Whether leverage the last bit of symmetric dtype. 
+                                            Defaults to False.
+                compression_factor (str, optional): Path of weight_config.json. Defaults to 32.
+                compression_dim (str, optional): Path of weight_config.json. Defaults to 'K'.
```
Comment (hshen14) [inline neural_compressor/model/torch_model.py:408]: also explain K is input channel of weight

File: neural_compressor/model/torch_model.py (hunk 2 above)
Comment (hshen14) [reply neural_compressor/model/torch_model.py:405]: Whether to leverage the full compression range under symmetric quantization

File: neural_compressor/model/torch_model.py (hunk 3 above)
Comment (hshen14) [reply neural_compressor/model/torch_model.py:407]: compressed_dtype

File: docs/source/quantization_weight_only.md [hunk 6]
```diff
@@ -73,7 +73,7 @@ q_model = quantization.fit(model, conf, eval_func=eval_func)
 q_model.save('saved_results')
 ```
 
-The saved_results folder contains two files: `best_model.pt` and `weight_config.json`, and the generated q_model is a fake quantized model.
+The saved_results folder contains two files: `best_model.pt` and `qweight_config.json`, and the generated q_model is a fake quantized model.
```
Comment (hshen14) [inline docs/source/quantization_weight_only.md:76]: is qweight.config (in json) better? how about pt q config file?

Comment (xin3he) [reply docs/source/quantization_weight_only.md:76]: we can wait for a proposal from outside when this feature becomes popular.

File: neural_compressor/adaptor/torch_utils/weight_only.py [hunk 7]
```diff
@@ -86,7 +86,7 @@ def qdq_weight_sym(weight, num_bits=4, quantile=1.0, return_int=False, full_rang
         minq = torch.tensor(2 ** (num_bits - 1) - 1)
     max_val = torch.max(weight, 1)[0]
     min_val = torch.min(weight, 1)[0]
-    flip_flag = torch.abs(min_val) > torch.abs(max_val)
+    flip_flag = torch.abs(max_val) > torch.abs(min_val)
     wmax = torch.max(torch.abs(max_val), torch.abs(min_val))
```
Comment (wenhuach21) [inline neural_compressor/adaptor/torch_utils/weight_only.py:90]: it's a little difficult to understand. why not just using code like this mapping_max = torch.abs(max_value)>torch.abs(min_value)? max_value: min_value, scale = mapping_max/-8. Maybe there are some bugs in my code ~

**Review by wenhuach21 — APPROVED** [review]

File: neural_compressor/adaptor/torch_utils/weight_only.py (hunk 7 above)
Comment (xin3he) [reply neural_compressor/adaptor/torch_utils/weight_only.py:90]: You're right! I also want to simplify this code, your proposal is promising, and I will have a try later.

**Pull request merged**